No java util date rule test

FanJups opened the pull request:
## Summary

<!-- Provide a general summary description of your changes -->

java.util.Date is [deprecated](https://www.math.uni-hamburg.de/doc/java/tutorial/post1.0/converting/deprecated.html)  so I decided to add a new rule **NoJavaUtilDateRuleTest.java** and suggested other libraries : 

1. java.time ( java 8)

1. java.util.GregorianCalendar

1.  java.Text.DateFormat

## Details

<!-- Describe your changes in detail -->

1) I created a new rule  file **src/main/java/ com.societegenerale.commons.plugin.rules.NoJavaUtilDateRuleTest.java** and added a new String constant (**NO_JAVA_UTIL_DATE_VIOLATION_MESSAGE**) in **src/main/java/com.societegenerale.commons.plugin.utils.ArchUtils.java**

2) I created **src/test/java/com.societegenerale.commons.plugin.rules.NoJavaUtilDateRuleTestTest.java** to test the rule and changed **NumberOfRulesAvailable from 9 to 10** & **totalClassesInPathWhenPackageFolderDoesNotExists from 12 to 13** in **src/test/java/com.societegenerale.commons.plugin.rules.ArchUtilsTest.java**

3) I created 3 files **ObjectWithJavaTextDateFormat.java ; ObjectWithJavaUtilDateReferences.java and ObjectWithJavaUtilGregorianCalendar.java** in **src/test/java/com.societegenerale.commons.plugin.rules.classesForTests**

4) I added my email in pom.xml

## Context

<!-- Why is this change required? What problem does it solve? -->

1. java.util.Date library is deprecated

1. Use Java8 java.time or java.util.GregorianCalendar or java.text.DateFormat  to parse and format dates instead of java.util.Date library because they  support internationalization better

<!-- Which environnment is/are impacted ? -->

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x ] I've added tests for my code.
- [x ] Documentation has been updated accordingly to this PR


## Related issue : 
<!-- If it fixes an open issue, please link to the issue here. -->

FanJups commented on src/main/java/com/societegenerale/commons/plugin/rules/NoJavaUtilDateRuleTest.java:
```diff
@@ -0,0 +1,86 @@
+							ArchUtils.NO_JAVA_UTIL_DATE_VIOLATION_MESSAGE + " - class: "
+									+ methodCall.getOriginOwner().getName() + " - line: "
+									+ methodCall.getLineNumber()));
+				}
```
It's a pleasure to contribute to this project. The source code format is automatically defined by Eclipse. I will also focus on these tests. Thanks for your idea.

FanJups commented on src/main/java/com/societegenerale/commons/plugin/rules/NoJavaUtilDateRuleTest.java:
```diff
@@ -0,0 +1,86 @@
+				List<JavaField> classesWithJavaUtilDateFields = item.getAllFields().stream()
+						.filter(field -> isJavaUtilDateField(field)).collect(toList());
+
+				for (JavaField field : classesWithJavaUtilDateFields) {
```
Great

FanJups commented on src/test/java/com/societegenerale/commons/plugin/rules/NoJavaUtilDateRuleTestTest.java:
```diff
@@ -0,0 +1,60 @@
+	}
+
+	@Test(expected = Throwable.class)
+	public void shouldThrowNoJavaUtilDateViolation() {
```
You are right, I'll delete this second test.

FanJups commented:
Thank you very much,what do you think about it ?

vincent-fuchs commented on src/main/java/com/societegenerale/commons/plugin/rules/NoJavaUtilDateRuleTest.java:
```diff
@@ -0,0 +1,76 @@
+						});
+			}
+
+			@SuppressWarnings("deprecation")
```
remove the SuppressWarnings annotation and modify your code so that no deprecation warning comes up.

vincent-fuchs commented:
> Thank you very much @vincent-fuchs , what do you think about the changes I've made ?

Thanks for the changes, but there is still one thing I don't get : 

- I still don't see a test for https://github.com/societe-generale/arch-unit-maven-plugin/blob/1051de70120f651afa80365e2789bd46f5325590/src/main/java/com/societegenerale/commons/plugin/rules/NoJavaUtilDateRuleTest.java#L54-L61 . I can still remove them, and no test is failing.

Also : 

- I've added a comment : https://github.com/societe-generale/arch-unit-maven-plugin/pull/5/files#r285387079 

- I've pushed a simplified version of NoJavaUtilDateRuleTestTest.java in the branch

- we've just pushed some fairly big changes in master : your branch needs to be rebased and conflicts fixed.

FanJups commented:
@vincent-fuchs I applied new  **changes related to : NoJavaUtilDateRuleTest.java / NoJavaUtilDateRuleTestTest.java / ArchUtilsTest.java / ObjectWithAdateField.java / ObjectWithJavaUtilDateReferences.java**

**NoJavaUtilDateRuleTest.java :** 

1. No method is imported using the java.util.Date library so I deleted code related to the cheking of imported methods

1. I created a constant  NO_JAVA_UTIL_DATE_VIOLATION_MESSAGE

1. I used SRC_CLASSES_FOLDER from ArchRuleTest

1. SuppressWarnings annotation is deleted

**NoJavaUtilDateRuleTestTest.java :**  I deleted **ObjectWithJavaUtilDateReferences.java**  and created **ObjectWithAdateField.java**

**ArchUtilsTest.java :** I changed NumberOfRulesAvailable from 9 to 10 & totalClassesInPathWhenPackageFolderDoesNotExists from 16 to 17

vincent-fuchs commented:
Thanks @FanJups ! We'll do a release soon, you'll be able to use it directly

FanJups commented:
Thanks @vincent-fuchs , that's great
